plugins/catalog-backend-module-bitbucket-server/src/lib/BitbucketServerClient.ts

```diff
@@ -81,7 +82,14 @@ export class BitbucketServerClient {
       request,
       getBitbucketServerRequestOptions(this.config),
     );
-    return response.json();
+    try {
+      const json_res = await response.json();
+      return json_res;
+    } catch {
+      throw new NotAllowedError(
+        `Malformed json response when attempting to get repository '${options.repo}' in project '${options.projectKey}' from Bitbucket Server. Repository might not exist.`,
+      );
```

Comment:
Does it really need to be malformed JSON? And why `NotAllowedError` for malformed JSON?
You don't really catch a specific error and validate it. There could be different errors here.

Insufficient permissions might be one. Or "Not Found" as your last sentence in the error message.

Overall, this seems a bit confusing to me.

Reply:
I think you are correct, not found would be better. I only included the "malformed JSON" because that was in the original error message. The only reason an error should be thrown on this part is if the given repository can't be found.